src/main/kotlin/com/ing/dlt/zkkrypto/ecc/schnorr/SchnorrSignature.kt

```diff
@@ -0,0 +1,162 @@
+package com.ing.dlt.zkkrypto.ecc.schnorr
+
+import com.ing.dlt.zkkrypto.ecc.EllipticCurve
+import com.ing.dlt.zkkrypto.ecc.EllipticCurvePoint
+import com.ing.dlt.zkkrypto.ecc.curves.AltBabyJubjub
+import com.ing.dlt.zkkrypto.ecc.schnorr.FixedGenerators.altBabyJubjubFixedGenerators
+import com.ing.dlt.zkkrypto.util.asUnsigned
+import org.bouncycastle.crypto.digests.Blake2bDigest
+import org.bouncycastle.crypto.digests.Blake2sDigest
+import java.math.BigInteger
+import java.security.SecureRandom
+import kotlin.random.Random.Default.nextBytes
+
+enum class Hash{
+    BLAKE2S,
+    BLAKE2B
+}
+
+/**
+ * Implementation of Schnorr signatures
+ * This implementation uses AltBabyJubJub curve and Blake2 hash
+ */
+
+class SchnorrSignature (
+    val curve: EllipticCurve,
+    val base: EllipticCurvePoint,
+    val compPersonalization: ByteArray,
+    val msgPersonalization: ByteArray) {
+
+    companion object {
+        fun zinc() = SchnorrSignature(
+            curve = AltBabyJubjub,
+            base = altBabyJubjubFixedGenerators()[5],
+            compPersonalization = "Zcash_RedJubjubH".toByteArray(),
+            msgPersonalization = "Matter_H".toByteArray()
+        )
+
+        const val MAX_MESSAGE_SIZE = 32
+        const val BLAKE2B_DIGEST_LENGTH = 64
+        const val BLAKE2S_DIGEST_LENGTH = 32
+    }
+
+    private lateinit var privateKey: BigInteger
```

Comment:
This is arguable. Consider this
```
class SchnorrSignature {
  ...
  val privateKey: BigInteger
  val publicKey: BigInteger
   
  init {
    nextKeyPair()
  }

  fun nextKeyPair() {
    privateKey = BigInteger(curve.S.bitLength(), SecureRandom())
        while (privateKey > curve.S) {
            privateKey = BigInteger(curve.S.bitLength(), SecureRandom())
        }
      publicKey = base.scalarMult(privateKey)
  }
   ...
}
```
then you won't need to do 
```
schnorr.generatePrivateKey()
        schnorr.getPublicKeyFromPrivate()
```
after each instantiation and can call `nextKeyPair` whenever needed (as you do in a loop in one of the tests)